src/typing/buildMacro.ml

```diff
@@ -0,0 +1,148 @@
+open Ast
+open Common
+open Type
+open Typecore
+open Error
+
+
+type candidate =
+	| Candidate of pos * path * string * recover_mode
+
+and recover_mode =
+	| RMBoth
+	| RMMethod
+	| RMModule
+	| RMNo
+
+
+let candidate_pos  = function
+	| Candidate(p,_,_,_) -> p
+
+let apply_macro ctx mode path el p =
+	let rev_path = List.rev path in
+	let starts_with_uppercase s =
```

Comment:
There's already `starts_uppercase` in path.ml. However, your implementation correctly checks the length of the String. We may want to remove the path.ml implementation because it's not safe to call.

On the other hand, your implementation is actually more like a `doesnt_start_with_lowercase`, which technically isn't quite the same.